server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -623,8 +638,17 @@ private void setFlushAfterMergeThresholdSize(ByteSizeValue byteSizeValue) {
         this.flushAfterMergeThresholdSize = byteSizeValue;
     }
 
+    private void setTranslogPruningByRetentionLease(boolean enabled) {
+        this.translogPruningByRetentionLease = enabled;
+        if(enabled) {
+            setTranslogRetentionSize(DEFAULT_TRANSLOG_RETENTION_SIZE);
+        }
+    }
+
     private void setTranslogRetentionSize(ByteSizeValue byteSizeValue) {
-        if (shouldDisableTranslogRetention(settings) && byteSizeValue.getBytes() >= 0) {
+        if (shouldDisableTranslogRetention(settings) &&
+            !shouldPruneTranslogByRetentionLease(settings) &&
```

Comment:
Why are we changing this logic?

Reply:
This is to ensure, we are keeping the existing behavior as is and only change based on translog pruning setting